File: 1/Solution.scala
/**
  Solution to project Eurler #1
  */
def sum3Or5(n: Int): Int = {
  (1 until n).filter(x => x%3 == 0 || x%5 == 0).sum
}

assert(sum3Or5(10) == 23)

println(sum3Or5(1000))

File: 1/solution.rs
fn main() {
    assert_eq!(23, sum3_or_5(10));
    println!("{}", sum3_or_5(1000).to_string());
}

fn sum3_or_5(n: u32) -> u32 {
    (1..n).filter(|x| x % 3 == 0 || x % 5 == 0).fold(0, |sum, x| sum + x)
}
